### PR TITLE
Utilize charges as additional anchor points for vampire drain dashboard

### PR DESCRIPTION
**Pull request by tacotran**
This change will additionally track vampire drain periods that include charges, including periods between drives and charges, charges and drives, and between two charges.

Partial solution to https://github.com/adriankumpf/teslamate/issues/429 although it does not exclude periods where the car is preheating.

### CODE REVIEW

File: grafana/dashboards/vampire-drain.json [hunk 1]
```diff
@@ -358,7 +358,7 @@
...
...nvert_km(CASE WHEN has_reduced_range THEN NULL ELSE (v.start_range - v.end_range) END, '$length_unit') AS range_diff_$length_unit,\n  CASE WHEN has_reduced_range THEN NULL ELSE (v.start_range - v.end_range) * c.efficiency END AS consumption,\n  CASE WHEN has_reduced_range THEN NULL ELSE ((v.start_range - v.end_range) * c.efficiency) / (v.duration / 3600) * 1000 END as avg_power,\n  convert_km(CASE WHEN has_reduced_range THEN NULL ELSE (v.start_range - v.end_range) / (v.duration / 3600) END, '$length_unit') AS range_lost_per_hour_[[length_unit]]\nFROM v, c,\n  LATERAL (\n    SELECT EXTRACT(EPOCH FROM sum(age(s.end_date, s.start_date))) as sleep\n    FROM states s\n    WHERE\n      state = 'asleep' AND\n      v.start_date <= s.start_date AND s.end_date <= v.end_date AND\n      s.car_id = $car_id\n  ) s_asleep,\n  LATERAL (\n    SELECT EXTRACT(EPOCH FROM sum(age(s.end_date, s.start_date))) as sleep\n    FROM states s\n    WHERE\n      state = 'offline' AND\n      v.start_date <= s.start_date AND s.end_date <= v.end_date AND\n      s.car_id = $car_id\n  ) s_offline\nWHERE\n  v.duration > ($duration * 60 * 60)\n  AND v.start_range - v.end_range >= 0\n  AND v.end_km - v.start_km < 1;\n",
```
[the remark this answers is not in the document]
Comment (tacotran) [reply grafana/dashboards/vampire-drain.json:361]: Thanks, I just realized that as well. 

Also https://github.com/adriankumpf/teslamate/commit/3a7eae4760ff4c04ec7287e333dd5b85b121d82a would probably break it even if it did work.

Let me go back to the drawing board. 

Unfortunately the date timestamps on the charging_processes table don't always exactly match up with what's in the positions table.

[the remark this answers is not in the document]
Comment (adriankumpf) [reply grafana/dashboards/vampire-drain.json]: > Also 3a7eae4 would probably break it even if it did work.

Good point. I hadn't even thought of that.

[the remark this answers is not in the document]
Comment (adriankumpf) [reply grafana/dashboards/vampire-drain.json]: I pushed 23f7c9c. Hope that fixes that. Let me know if you spot any errors.

[the remark this answers is not in the document]
Comment (tacotran) [reply grafana/dashboards/vampire-drain.json]: I didn't even think of moving the SOC query up! I was working on a way to join the charges table on to my original method, but this is much cleaner.

Looks great!

**Pull request merged**